src/whir/committer/mod.rs

```diff
@@ -17,14 +30,42 @@ pub struct Witness<F, MerkleConfig>
 where
     MerkleConfig: Config,
 {
-    /// The committed polynomial in coefficient form.
+    /// The committed polynomial in coefficient form. In case of batching, its
+    /// the batched polynomial, i.e., the weighted sum of polynomials in
+    /// batching data.
     pub(crate) polynomial: CoefficientList<F>,
-    /// The Merkle tree constructed from the polynomial evaluations.
+
+    /// The Merkle tree constructed from the polynomial evaluations. In case of
+    /// batching, it's the merkle tree of the batched polynomial.
     pub(crate) merkle_tree: MerkleTree<MerkleConfig>,
-    /// The leaves of the Merkle tree, derived from folded polynomial evaluations.
+
+    /// The leaves of the Merkle tree, derived from folded polynomial
+    /// evaluations. In case of batching, its the merkle leaves of the batched
+    /// tree. These leaves are computed as the weighted sum leaf values in the
+    /// batching_data.
     pub(crate) merkle_leaves: Vec<F>,
+
     /// Out-of-domain challenge points used for polynomial verification.
     pub(crate) ood_points: Vec<F>,
+
     /// The corresponding polynomial evaluations at the OOD challenge points.
     pub(crate) ood_answers: Vec<F>,
+
+    /// The individual commitments of each batching polynomial
+    pub(crate) batching_data: Vec<BatchingData<F, MerkleConfig>>,
+
+    /// The batching randomness. If there's no batching, this value is zero.
+    pub(crate) batching_randomness: F,
```

Comment:
This can become a Vec<F>, in the batching_1 this will be set to [F::ONE], otherwise it's [F::ONE, r, r^2, ...] ecc.